### DISCUSSION TITLE
Should I Rollback a transaction on error explicitly ?

### DISCUSSION THREAD

**Discussion by youssefbennour** [ServiceStack/Discuss#84] [Q&A] [answered]
Hello,
I've been curious about, if I should rollback tranasctions explicitly in catch blocks (on Exceptions) or the implemenation of IDisposable would manage that instead.
As Microsoft docs  (https://learn.microsoft.com/en-us/dotnet/api/system.data.common.dbtransaction.dispose?view=net-7.0) : 
"Dispose should rollback the transaction. However, the behavior of is provider specific, and should not replace calling".
I'm using the SqlServerDialect provider, I'm curious about whether the implementation you're using has the above feature.

**Comment by mythz** [comment, marked as answer]
All your support questions appear to be easily answered with reading docs or doing a bit of your own research, this forums shouldn't be a a substitute for doing your own research, of which being self-taught is probably the single most important skill for becoming a good developer.

All .NET ORMs us the same SQL Server ADO.NET providers so the behavior isn't specific to OrmLite, have you tried writing a simple test to find out what the behavior is? Otherwise you can just follow Microsoft docs which are correct.

**Reply by youssefbennour** [reply]
Yes, that's what I should've done.
Altough I've done some research but I found no direct answer, but I should've tested that by myself.
I'm sorry.
